contract/src/platform-goals/start-governed-contract.js

```diff
@@ -0,0 +1,280 @@
+  const committeeName = `${contractName}Committee`;
+  const charterName = `${contractName}Charter`;
+
+  const {
+    consume: { namesByAddress, zoe },
+    produce: { [`${charterName}Kit`]: produceKit },
+    installation: {
+      consume: { binaryVoteCounter: counterP, econCommitteeCharter: installP },
+    },
+    instance: {
+      produce: { [charterName]: instanceP },
+    },
+  } = powers;
+  const {
+    [committeeName]: { voterAddresses },
+  } = config?.options || {};
+  assert(voterAddresses, 'no voter addresses???');
+
+  const [charterInstall, counterInstall] = await Promise.all([
+    installP,
+    counterP,
+  ]);
+  const terms = await allValues({
+    binaryVoteCounterInstallation: counterInstall,
+  });
+
+  const startResult = await E(zoe).startInstance(
+    charterInstall,
+    undefined,
+    terms,
+    undefined,
+    'econCommitteeCharter',
+  );
+  instanceP.resolve(startResult.instance);
+  produceKit.resolve(startResult);
+
+  await inviteToMyCharter(
+    startResult.creatorFacet,
+    namesByAddress,
+    voterAddresses,
+  );
+  return startResult;
+};
+harden(startMyCharter);
+
+/**
+ * @param {string} contractName
+ * @param {BootstrapPowers} powers
+ * @param {*} config
+ */
+export const startMyCommittee = async (contractName, powers, config) => {
+  const committeeName = `${contractName}Committee`;
+  const {
+    consume: { board, chainStorage, namesByAddress, startUpgradable },
+    produce: { [`${committeeName}Kit`]: produceKit },
+    installation: {
+      consume: { committee },
+    },
+    instance: {
+      produce: { [committeeName]: produceInstance },
+    },
+  } = powers;
+  const {
+    [committeeName]: { voterAddresses },
+  } = config?.options || {};
+  assert(voterAddresses, 'no voter addresses???');
+
+  const committeesNode = await E(chainStorage).makeChildNode(COMMITTEES_ROOT);
+  const storageNode = await E(committeesNode).makeChildNode(
+    sanitizePathSegment(committeeName),
+  );
+  const marshaller = await E(board).getPublishingMarshaller();
+
+  const privateArgs = {
+    storageNode,
+    marshaller,
+  };
+
+  const started = await E(startUpgradable)({
+    label: committeeName,
+    installation: await committee,
+    terms: { committeeName, committeeSize: values(voterAddresses).length },
+    privateArgs,
+  });
+  produceKit.resolve(started);
+  produceInstance.resolve(started.instance);
+  console.log(committeeName, 'started');
+
+  /** @param {[string, Promise<Invitation>][]} addrInvitations */
+  const distributeInvitations = async addrInvitations => {
+    await Promise.all(
+      addrInvitations.map(async ([addr, invitationP]) => {
+        const debugName = `econ committee member ${addr}`;
+        const depositFacet = E(namesByAddress).lookup(addr, 'depositFacet');
+        await E.when(invitationP, invitation =>
+          E(depositFacet).receive(invitation),
+        ).catch(err => console.error(`failed deposit to ${debugName}`, err));
```

Comment:
I tried adding an address through the env variable `SWAP_GOV_ADDR`  in the Makefile which seemed to successfully pipe through to this code, but I'm seeing an error here when deploying the contract:

`"nameKey" not found: "agoric1rwwley550k9mmk6uq6mm6z4udrg8kyuyvfszjk"`

This is confusing because in the logs I can see that this address has a smart wallet and is opening vaults before it gets to this point. Is there something wrong with `namesByAddress` in this context?

Reply:
Maybe you're running on a chain that doesn't have the fix for this one yet? I don't recall what release it's in.

 - https://github.com/Agoric/agoric-sdk/issues/8113